app/models/relationship.rb

```diff
@@ -159,6 +159,13 @@ def create_or_update_parent_id
   def self.create_unless_exists(source_id, target_id, relationship_type, options = {})
     # Verify that the target is not part of another source; in this case, we should return the existing relationship.
     r = Relationship.where(target_id: target_id).where.not(source_id: source_id).last
+    # should update options if exists and at least on field is blank
+    if !r.nil? && !options.blank? && r.model.blank?
```

Comment:
Sawy, as we discussed in the meeting, please make sure that `options` contains at least one non-blank value.